WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressTopBannerFactory.swift

```diff
@@ -5,17 +5,37 @@ import Yosemite
 ///
 final class ShippingLabelAddressTopBannerFactory {
     static func addressErrorTopBannerView(shipType: ShipType,
+                                          phoneNumber: String?,
                                           openMapPressed: @escaping () -> Void,
                                           contactCustomerPressed: @escaping () -> Void) -> TopBannerView {
         // Set banner text and action buttons based on shipping address type (origin or destination).
-        let infoText = shipType == .destination ? Localization.infoShipTo : Localization.infoShipFrom
+        let missingPhoneNumber = phoneNumber == nil || phoneNumber?.isEmpty == true
```

Comment:
I'd want to consider a nil `phoneNumber` as missing too (this happens when address is not found). So I think this check covers more edge cases - that makes sense right?

Reply:
Oh, yes! I just meant to ask whether the `isEmpty` check can be simplified. So instead of this:

`let missingPhoneNumber = phoneNumber == nil || phoneNumber?.isEmpty == true`

Can that line be this:

`let missingPhoneNumber = phoneNumber == nil || phoneNumber?.isEmpty`

Reply:
@rachelmcr since `phoneNumber?.isEmpty` is an optional boolean, I have to explicitly check with ` == true`, otherwise the compiler will be confused and complain something like "Optional type 'Bool?' cannot be used as a boolean; test for '!= nil' instead". So that check is necessary 😄

Reply:
Ah, right! Now I get it; thanks for the explanation :)